core/src/test/java/org/elasticsearch/index/translog/SnapshotMatchers.java

```diff
@@ -128,4 +136,51 @@ public void describeTo(Description description) {
     }
 
 
+    public static class ContainingInAnyOrderMatcher extends TypeSafeDiagnosingMatcher<Translog.Snapshot> {
+        private final Collection<Translog.Operation> expectedOps;
+
+        static List<Translog.Operation> drainAll(Translog.Snapshot snapshot) throws IOException {
+            final List<Translog.Operation> actualOps = new ArrayList<>();
+            Translog.Operation op;
+            while ((op = snapshot.next()) != null) {
+                actualOps.add(op);
+            }
+            return actualOps;
+        }
+
+        public ContainingInAnyOrderMatcher(Collection<Translog.Operation> expectedOps) {
+            this.expectedOps = expectedOps;
+        }
+
+        @Override
+        protected boolean matchesSafely(Translog.Snapshot snapshot, Description mismatchDescription) {
+            try {
+                List<Translog.Operation> actualOps = drainAll(snapshot);
+
+                List<Translog.Operation> notFound = expectedOps.stream()
+                    .filter(o -> actualOps.contains(o) == false)
+                    .collect(Collectors.toList());
+                if (notFound.isEmpty() == false) {
+                    mismatchDescription
+                        .appendText(" Operations not found").appendValueList("[", ", ", "]", notFound);
+                }
+
+                List<Translog.Operation> notExpected = actualOps.stream()
+                    .filter(o -> expectedOps.contains(o) == false)
+                    .collect(Collectors.toList());
+                if (notExpected.isEmpty() == false) {
+                    mismatchDescription
+                        .appendText(" Operations not expected ").appendValueList("[", ", ", "]", notExpected);
+                }
+                return notFound.isEmpty() && notExpected.isEmpty();
+            } catch (IOException ex) {
+                throw new ElasticsearchException("failed to read snapshot content", ex);
+            }
+        }
+
+        @Override
+        public void describeTo(Description description) {
+
```

Comment:
Should this be empty?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/27268/commits/ebe92b665dfbfda3f355cfa83f619b8e835d9d5d. Thank you.